### PR TITLE
Fix excessive text elision workbench data view

### LINKED ISSUE #34101: Workbench: Show Data doesn't show values until column resized
**Describe the bug**

Show Data on a workspace no longer shows the values until a column is resized. See screenshot

**To Reproduce**

1. Load any data
2. Click show data

**Expected behavior**

The values are visible without resizing the column.

**Screenshots**

image

**Platform/Version (please complete the following information):**
 - OS: Windows
 - OS Version: 10
 - Mantid Version: 6.4+

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

This work addresses the issues with excessive text elision in the matrixworkspace data view after upgrading to Qt 5.12. See the screenshot in the linked issue for a comparison where the old version is on Qt 5.9.

It is not clear that this is a bug in Qt, just a change in behaviour. Here we introduce a delegate to control the amount of elision of the string that occurs.

**To test:**

* Open workbench
* Load any data file
* Show the data and most digits after the decimal point should be shown. Widen the column to show everything.
* Also check the background colours are respected. Run `MaskDetectors` and mask out several rows.
  Open the data view again and see the masked rows have a grey background color
* Check the monitor background is colored yellow. Run `LoadEmptyInstrument` with `InstrumentName=IRIS`. The first two rows in the data view should have a yellow background

Fixes #34101

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqt/mantidqt/widgets/workspacedisplay/matrix/delegate.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
```
Comment (sf1919) [inline qt/python/mantidqt/mantidqt/widgets/workspacedisplay/matrix/delegate.py:3]:
```suggestion
# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
```

File: qt/python/mantidqt/mantidqt/widgets/workspacedisplay/matrix/test/test_matrixworkspacedisplay_delegate.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
```
Comment (sf1919) [inline qt/python/mantidqt/mantidqt/widgets/workspacedisplay/matrix/test/test_matrixworkspacedisplay_delegate.py:3]:
```suggestion
# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
```

**Review by sf1919 — CHANGES_REQUESTED** [review]
I have checked the functionality and it works well under all circumstances (including line shading). Happy with the majority of the code - just one or two minor changes requested.

**Comment by martyngigg** [comment]
Thanks for picking up those typos. They should be fixed now.

**Review by sf1919 — APPROVED** [review]

**Pull request merged**